84.md

```diff
@@ -0,0 +1,98 @@
+
+## Format
+The `.content` of these events is the highlighted portion of the text.
+
+`.content` might be empty for highlights of non-text based media (e.g. NIP-94 audio/video).
+
+### References
+Events SHOULD tag the source of the highlight, whether nostr-native or not.
+`a` or `e` tags should be used for nostr events and `r` tags for URLs.
+
+When tagging a URL, clients generating these events SHOULD do a best effort of cleaning the URL from trackers
+or obvious non-useful information from the query string.
+
+### Attribution
+Clients MAY include one or more `p` tags, tagging the original authors of the material being highlighted; this is particularly
+useful when highlighting non-nostr content for which the client might be able to get a nostr pubkey somehow
+(e.g. prompting the user or reading a `<meta name="nostr:pubkey" />` tag on the document). A role MAY be included as the
+last value of the tag.
+
+```json
+[ "p", "...", "author" ],
+[ "p", "...", "author" ],
+[ "p", "...", "editor" ],
+```
+
+### Context
+Clients MAY include a `context` tag, useful when the highlight is a subset of a paragraph and displaying the
+surrounding content might be beneficial to give context to the higlight.
+
+### Ranges
+Clients MAY include `range` tags with the start/end indexes of where the highlight begins and finishes within
+the referenced article/tagged-event.
+
+```
+[ "range", <start-index>, <end-index> ]
+```
+
+Additionally a range with `context` as the third value of the tag MAY be added to indicate the begin/finish indexes
+of the highlight within the included `context` tag.
+
+```
+[ "range", <start-index>, <end-index>, "context" ]
+```
+
+#### Text-based nostr events' highlights
+
+Highlights of nostr-natives events SHOULD use the range index of the content as-is
+(e.g. NIP-23 articles include the markdown instead of computing the index from the rendered markdown).
+
+```
+[ "range", 3000, 3042 ] # highlight begins at index position 3000 of the tagged event's `.content`
+[ "range", 42, 84, "context" ] # highlight begins at index position 42 of the `context` tag's value
+```
+
+#### Non-text-based nostr events's highlights
+
+A `kind:9802` event that tags a NIP-94 event which includes a video or audio file can use ranges to
+indicate the start/end time position in seconds.
+
+#### Ranges in URL highlights
+
+When creating a highlight from a URL the range should be expressed over
+the extracted plain text of the formatted content (e.g. rendered HTML instead of including the HTML markup);
+this helps make finding the correct indexes easier on websites with markup variations on each render.
+
+e.g. `<html><body>hello, world</body></html>`
+
+Tagging `hello, world` would result in using a range tag like `["range", 0, 12 ]`.
+
+Text extraction (i.e. translation from non-plain text medium like HTML or PDF) is highlighy subjective and the value
+of the range should be carefully interpreted by the different clients that support this.
```

Comment:
Have you found this to be fairly reliable in practice?

Reply:
Yeah, there are a handful of libraries that do this text extraction from the DOM, and they seem to be fairly consistent. The index in this case could be thought of more as an approximate value.